src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -97,213 +92,406 @@ Position const SpawnBat[6] =
+            me->SetSpeed(MOVE_WALK, 5.0f, true);
+
+            // start the flight loop
+            me->SetCanFly(true);
+            me->SetDisableGravity(true);
+            me->GetMotionMaster()->MoveSplinePath(PATH_BATRIDER_LOOP);
+        }
+        // otherwise, trash mode
+        else
+        {
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BATRIDER_MODE_TRASH");
+            _mode = BATRIDER_MODE_TRASH;
+
+            me->SetReactState(REACT_DEFENSIVE);
+
+            // don't interrupt casting
+            _scheduler.SetValidator([this]
+            {
+                return !me->HasUnitState(UNIT_STATE_CASTING);
+            });
+        }
     }
 
     void Reset() override
     {
+        CreatureAI::Reset();
+
         _scheduler.CancelAll();
-        me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-        me->SetHover(true);
-        me->SetDisableGravity(true);
-        me->AddUnitState(UNIT_STATE_ROOT);
+
+        if (_mode == BATRIDER_MODE_BOSS)
+        {
+            me->GetMotionMaster()->Clear();
+        }
+        else if (_mode == BATRIDER_MODE_TRASH)
+        {
+            // apply the Thrash (8876) spell to the bat rider (passive ability)
+            me->CastSpell(me, SPELL_BATRIDER_PASSIVE_THRASH);
+        }
     }
 
-    void JustEngagedWith(Unit* /*who*/) override
+    void JustEngagedWith(Unit* who) override
     {
-        _scheduler.Schedule(2s, [this](TaskContext context)
+        CreatureAI::JustEngagedWith(who);
+
+        if (_mode == BATRIDER_MODE_BOSS)
         {
-            DoCastRandomTarget(SPELL_THROW_LIQUID_FIRE);
-            context.Repeat(7s);
-        });
+            // throw bomb
+            _scheduler.Schedule(2s, [this](TaskContext context)
+            {
+                DoCastRandomTarget(SPELL_BATRIDER_THROW_LIQUID_FIRE);
+                context.Repeat(8s);
+            });
+        }
+        else if (_mode == BATRIDER_MODE_TRASH)
+        {
+            // Demo Shout
+            _scheduler.Schedule(1s, [this](TaskContext /*context*/)
+            {
+                DoCastSelf(SPELL_BATRIDER_DEMO_SHOUT);
+            // Battle Command
+            }).Schedule(8s, [this](TaskContext context)
+            {
+                DoCastSelf(SPELL_BATRIDER_BATTLE_COMMAND);
+                context.Repeat(25s);
+            // Infected Bite
+            }).Schedule(6500ms, [this](TaskContext context)
+            {
+                DoCastVictim(SPELL_BATRIDER_INFECTED_BITE);
+                context.Repeat(8s);
+            });
+        }
     }
 
-    void UpdateAI(uint32 diff) override
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType, SpellSchoolMask) override
     {
-        if (!UpdateVictim())
-            return;
-
-        _scheduler.Update(diff);
+        if (_mode == BATRIDER_MODE_TRASH)
+        {
+            // if health goes below 30%, cast unstable concoction
+            if (me->HealthBelowPctDamaged(30, damage))
```

Comment:
ScheduleHealthCheckEvent to remain consistent

Reply:
Not available in `CreatureAI`. Do you want me to convert to BossAI?